src/components/QrScan/QrScan.js

```diff
@@ -65,18 +73,78 @@ const QrScan = () => {
           setQrCodes(tempScannedCodes);
           history.push('/display-results');
         }
-
         setScannedCodes(tempScannedCodes);
       } else {
         setQrCodes([data]);
         history.push('/display-results');
       }
     }
-  }
+  };
 
   const handleError = () => {
     history.push('/display-results');
-  }
+  };
+
+  /**
+   * Create QrScanner instance using video element and specify result/error conditions
+   * @param {*} videoElement HTML video element
+   */
+  const createQrScanner = (videoElement) => {
+    if (!videoElement) {
+      if (runningQrScanner.current) {
+        qrScan.destroy();
+      }
+      return;
+    }
+
+    qrScan = new QrScanner(
+      videoElement,
+      (data) => {
+        setScannedData(data);
+      },
+      () => {},
+      () => ({
+        // define scan region for QrScanner
+        x: 0,
+        y: 0,
+        width: videoElement.videoWidth,
+        height: videoElement.videoHeight,
+      })
+    );
+    runningQrScanner.current = qrScan;
+    qrScan.start().then(() => {
+      qrScan.hasCamera = true;
+    });
+  };
+
+  const videoRef = useRef(null);
+  // Get user media when the page first renders, and feed into createQrScanner()
+  useEffect(() => {
+    const getUserMedia = async () => {
+      try {
+        const stream = await navigator.mediaDevices.getUserMedia({
+          video: true,
+        });
+        videoRef.current.srcObject = stream;
+      } catch (err) {
+        console.log(err);
```

Comment:
Lint gets annoyed when you use the console. The easiest way to fix this is probably to simply throw an error here. 

I'm not sure if there's a specific way we want to handle errors like this, so I would defer to others if people have thoughts, but something like this should work for now:

```suggestion
       throw Error('Cannot access video.')
```